From aaf584a3ec131dd990388db818db134651e6ac0a Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 11:59:15 -0400
Subject: [infiniband] ehca: Remove ib_sg_dma_address() and ib_sg_dma_len() overloads

Message-id: <bd94469168fd82eaf823298b8f6f7beb3c358594.1410425016.git.avadai@redhat.com>
Patchwork-id: 93591
O-Subject: [Patch RHEL7.1 V1 019/197] IB/ehca: Remove ib_sg_dma_address() and ib_sg_dma_len() overloads
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Mike Marciniszyn <mike.marciniszyn@intel.com>

These methods appear to only mimic the sg_dma_address() and
sg_dma_len() behavior.

They can be safely removed.

Suggested-by: Bart Van Assche <bvanassche@acm.org>
Cc: Bart Van Assche <bvanassche@acm.org>
Cc: Hoang-Nam Nguyen <hnguyen@de.ibm.com>
Cc: Christoph Raisch <raisch@de.ibm.com>
Reviewed-by: Dennis Dalessandro <dennis.dalessandro@intel.com>
Signed-off-by: Mike Marciniszyn <mike.marciniszyn@intel.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit f3585a6ae31cb0f0ebda53b161fbed7c9a679572)
Signed-off-by: Amir Vadai <amirv@mellanox.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/ehca/ehca_mrmw.c b/drivers/infiniband/hw/ehca/ehca_mrmw.c
index 7168f59..3488e8c 100644
--- a/drivers/infiniband/hw/ehca/ehca_mrmw.c
+++ b/drivers/infiniband/hw/ehca/ehca_mrmw.c
@@ -2534,16 +2534,6 @@ static void ehca_dma_unmap_sg(struct ib_device *dev, struct scatterlist *sg,
  /* This is only a stub; nothing to be done here */
 }
 
-static u64 ehca_dma_address(struct ib_device *dev, struct scatterlist *sg)
-{
- return sg->dma_address;
-}
-
-static unsigned int ehca_dma_len(struct ib_device *dev, struct scatterlist *sg)
-{
- return sg->length;
-}
-
 static void ehca_dma_sync_single_for_cpu(struct ib_device *dev, u64 addr,
       size_t size,
       enum dma_data_direction dir)
@@ -2596,8 +2586,6 @@ struct ib_dma_mapping_ops ehca_dma_mapping_ops = {
  .unmap_page             = ehca_dma_unmap_page,
  .map_sg                 = ehca_dma_map_sg,
  .unmap_sg               = ehca_dma_unmap_sg,
- .dma_address            = ehca_dma_address,
- .dma_len                = ehca_dma_len,
  .sync_single_for_cpu    = ehca_dma_sync_single_for_cpu,
  .sync_single_for_device = ehca_dma_sync_single_for_device,
  .alloc_coherent         = ehca_dma_alloc_coherent,
-- 
1.7.1